lib/integration/scenarios/Test/Integration/Scenario/API/Shelley/Restoration.hs

```diff
@@ -208,5 +213,5 @@ restoringFromCheckpoint
 restoringFromCheckpoint cp =
     setRestorationMode
         $ RestoreFromBlock
-            (headerHash cp)
-            (SlotNo $ getQuantity $ slotNo cp)
+            (Read.SlotNo $ fromIntegral $ getQuantity $ slotNo cp)
+            (fromJust $ Hash.hashFromBytes $ getHash $ headerHash cp)
```

Comment:
no chance to get rid of `fromJust` here?

Reply:
Fair enough.

I have added a function `toRawHeaderHash`. This function still needs to use `fromJust`, but it's not partial — at least as long as the invariant that values of type `Hash "BlockHeader"` are valid blake2b_256 hashes is respected.